torch/onnx/_internal/exporter.py

```diff
@@ -698,26 +696,37 @@ def __call__(
         Args:
             args: The positional inputs to the model.
             kwargs: The keyword inputs to the model.
-            model_with_state_dict: The PyTorch model to fetch state from.
-                Required when :func:`enable_fake_mode` is used to extract real initializers as needed by the ONNX graph.
+            exported_program: Required when :func:`enable_fake_mode` is used to extract real initializers as
+                needed by the ONNX graph. In fake mode export, constant tensors and non-persistent buffers
+                are lost, as they are not supported in state_dict, so the original model is needed to
+                fetch the initializers.
             options: The options to use for running the model with ONNX Runtime.
 
         Returns:
             The model output as computed by ONNX Runtime
         """
         import onnxruntime  # type: ignore[import]
 
-        # model specified by the user has precedence, when specified
-        model_with_state_dict = model_with_state_dict or self._model_torch
```

Comment:
@thiagocrepaldi I will revert the refactor here, but there are something we can think about I think. What purpose is torch.nn.Module serving as a parameter in this __call__ and adapt_torch_inputs_to_onnx when it doesn't provide anything (it doesn't have stat_dict and mdel_signature). Look on the previous code, nn.Module is just fitting here to compromise ExportedProgram usage. That's why I proposed exported_program specific parameter to de-confused. I really thought nn.Module did something in this function.